### PR TITLE
Added cluster 'proxy' functions

### PR DESCRIPTION
**Pull request by wewang58**
@openshift/devexp-qe please help to review it, thx

### CODE REVIEW

**Comment by xiuwang** [comment]
LGTM, thanks

File: lib/openshift/proxy_config_openshift_io.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+require 'openshift/cluster_resource'
+
+module BushSlicer
+  class ProxyConfigOpenshiftIo < ClusterResource
+    RESOURCE = 'proxies.config.openshift.io'
+
+    def httpproxy(user: nil, quiet: false, cached: true)
```
Comment (pruan-rht) [inline lib/openshift/proxy_config_openshift_io.rb]: @wewang58 convention for methods is to use underscore

File: lib/openshift/proxy_config_openshift_io.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      rr = raw_resource(user: user, cached: cached, quiet: quiet)
+      return rr.dig('spec', 'httpProxy')
+    end
+
+    def httpsproxy(user: nil, quiet: false, cached: false)
```
Comment (pruan-rht) [inline lib/openshift/proxy_config_openshift_io.rb]: @wewang58 use underscore for method names instead like `https_proxy`

**Pull request merged**